src/controller/controlador_produtos.py

```diff
@@ -102,15 +158,19 @@ def excluir_produto(self) -> Produto:
 
         self.__tela_produto.cadastro_nao_encontrado()
 
-    def pesquisa_produto(self, codigo: int) -> Produto:
+    def mostrar_erro(self, mensagem: str):
+        print(f"Erro: {mensagem}")
```

Comment:
```suggestion
    def mostrar_erro(self, mensagem: str):
        self.__tela_produto.mostrar_erro(mensagem)
```

Eu tinha esquecido de adicionar essa função no controlador de produtos. 
Quando usamos o @tratar_excecao ele sempre chama ela pra qualquer exceção levantada.